bump regression test pg14 to rc1

efourd opened the pull request:
bumps pg14 pgver to 14rc1.
Also changes diff output step to only run on failure of previous step.

Commit:
Bump regression test pg14rc1 to pg14

Commit:
Only run show regression diffs on failure

mpalmi commented on .github/workflows/regression-tests.yml:
```diff
@@ -25,7 +25,7 @@ jobs:
           - {pgver: 12, devpkg: 12}
           - {pgver: 13, devpkg: 13}
-          - {pgver: 14beta3, devpkg: 14}
+          - {pgver: 14, devpkg: 14}
```
Since these are all now, `pgver` == `devpkg`, can we do something like this and drop the verbose entries (L21-L28)?

```suggestion
          pgver: [9.4, 9.5, 9.6, 10, 11, 12, 13, 14]
```


and then in the Dockerfile: 
https://github.com/pgaudit/set_user/blob/master/.github/docker/Dockerfile#L6

`ARG DEVPKG=PGVER`

efourd replied:
Addressed in https://github.com/pgaudit/set_user/pull/54/commits/02d865a6851a9e95dbedad4afeed6c0a9eaa0d75

Commit:
Add DEVPKG env var to pass DEVPKG docker build arg

DEVPKG is set to pgver value unless a new devpkg value
is specified, example:
        include:
          - pgver: 14beta3
            devpkg: 14
